code/game/objects/structures/crates_lockers/closets/secure/guncabinet.dm

```diff
@@ -30,22 +52,199 @@
+
+	var/datum/browser/popup = new(user, "gunlocker", "<div align='center'>[name]</div>", 350, 300)
+	popup.set_content(dat)
+	popup.open(0)
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/Topic(href, href_list)
+	..()
+	if(href_list["retrieve"])
+		var/obj/item/O = locate(href_list["retrieve"]) in contents
+		if(!O || !istype(O))
+			return
+		if((!CanUseTopic(usr, GLOB.tgui_default_state)) || !opened)
+			return
+		if(ishuman(usr))
+			usr.visible_message("<span class='notice'>[usr] pulls [O] out of [src]!</span>", "<span class='notice'>You retrieve [O] from the locker.</span>")
+			if(!usr.put_in_hands(O))
+				O.forceMove(get_turf(src))
+			update_icon()
+		updateDialog()
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/update_icon()
+	cut_overlays()
+	if(LAZYLEN(contents))
+		for(var/i in 1 to contents.len)
+			var/obj/item/I = contents[i]	//we've already made sure there's only guns
+			var/mutable_appearance/gun_overlay = mutable_appearance(icon, I.icon_state)
+			gun_overlay.pixel_x = 3 * (i - 1)
+			add_overlay(gun_overlay)
+	if(welded)
+		add_overlay("[icon_state]_cut")
+		layer = OBJ_LAYER
+		return
+	else if(opened)
+		add_overlay("[icon_state]_open")
+		layer = OBJ_LAYER
+		return
+	else
+		add_overlay("[icon_state]_door")
+		if(broken)
+			add_overlay("[icon_state]_off")
+			add_overlay("[icon_state]_sparking")
+		else if(locked)
+			add_overlay("[icon_state]_locked")
+		else
+			add_overlay("[icon_state]_unlocked")
+		return
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/shotgun
+	name = "long arms locker"
+	icon_state = "shotguncase"
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/rifle
+	name = "long arms locker"
+	icon_state = "riflecase"
+	desc = "A strong cabinet used for securing firearms. This one is for long arms such as rifles and shotguns."
+
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/rifle/wood
+	icon_state = "riflefancy"
+
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/pistol
+	name = "small arms locker"
+	icon_state = "pistolcase"
+	desc = "A strong cabinet used for securing firearms. This one is for hand-held sidearms."
+	case_type = GUN_SIDEARM
+	capacity = 10
+
+/obj/structure/closet/secure_closet/guncabinet/fancy/pistol/update_icon()
+	cut_overlays()
+	var/gunrow = 0
+	var/gunmax = 5
+	var/racked = FALSE
+	if(LAZYLEN(contents))
+		for(var/i in 1 to contents.len)
+			gunrow++
+			if(contents.len <=5) //hopefully avoids layering issues + infinite rows (it doesnt)
+				racked = FALSE
+			if(gunrow >= gunmax && !racked) // start the next row of sprites atop of the previous row
+				racked = TRUE
+				gunrow = 0
+			var/obj/item/I = contents[i]	//we've already made sure there's only guns
+			var/mutable_appearance/gun_overlay = mutable_appearance(icon, I.icon_state)
+			if(racked)
+				gun_overlay.pixel_x = 3 * gunrow
+				gun_overlay.pixel_y = -1
+			else
+				gun_overlay.pixel_y = 1
+				gun_overlay.pixel_x = 3 * (gunrow - 1)
+			add_overlay(gun_overlay)
```

Comment:
This is me trying to be way too fucking clever for my own good, probably. the intent is to have two rows of overlays stacked after each other.

the only issue is not being updated properly + overhang of sprites. Frustrating but I just wanted to have something in place anyway. 

the pistol locker is *really* tiny, though, so it might just need to be made bigger and the pistol sprites larger than 5x4